streams/dlna_meta.py

```diff
@@ -1,68 +1,214 @@
+              fname = (i.text.split('/')[-1])[0:200]
+              if debug:
+                print(f"Downloading album art {i.text} to {album_art_dir}/{fname}")
+
+              # move to the album art directory and download the file (if it changed), if it fails delete the file
+              os.system(f"cd {album_art_dir} && wget -q -N {i.text} -O {fname} || rm -f {album_art_dir}/{fname}")
+
+              # if the file exists, set the metadata to the file name
+              if os.path.exists(f"{album_art_dir}/{fname}"):
+                if fname != last_file: # if it's a new file delete the old one
+                  if last_file: # handle the case where this is the first file downloaded
+                    os.remove(f"{album_art_dir}/{last_file}")
+                  last_file = fname
+
+                metadata["album_art"] = fname
+              else:
+                metadata["album_art"] = ""
+        except Exception as e:
+          print(f"Error: could not get song-info: {e}")
 
-# Check if the folder is valid and writeable
-if not os.path.exists(args.dlna_config_dir) or not os.access(args.dlna_config_dir, os.W_OK):
-  print('Error: unable to access: {}'.format(args.dlna_config_dir))
-
-# trim trailing /
-if args.dlna_config_dir[-1] == '/':
-  args.dlna_config_dir = args.dlna_config_dir[:-1]
-
-cs_loc = '{}/currentSong'.format(args.dlna_config_dir)
-cs_conf = {
-    'state': 'playing'
-}
-ctmd = re.compile(r'dc:(.*?)>(.*?)</dc:|upnp:(.*?)>(.*?)</upnp:')
-ts = re.compile(r'TransportState: ([A-Z]*)')
-
-def read_field():
-  line = sys.stdin.readline()
-  line = line.strip('\n')
-  if line[0:4] == "INFO":
-    s1 = line.split('] ')
-    return s1[1]
+        if debug:
+          print("writing metadata", metadata)
+
+        # empty file and write metadata
+        f.truncate(0)
+        f.seek(0)
+        json.dump(metadata, f)
+        f.flush()
+      except Exception as e:
+        print(f"Error: could not write metadata: {e}")
+
+
+# Blocks until a line is put in the fifo, then executes the command in that line.
+def command_executor(fifo_path: str, service: SSDPDevice, debug: bool = False):
+  """ Execute commands from the fifo file. """
+  # if fifo does not exist, create it
+  if not os.path.exists(fifo_path):
+    os.mkfifo(fifo_path)
+    if debug:
+      print(f"Created fifo file at {fifo_path}")
   else:
-    return None
-
-def meta_parser(fstring):
-  u = {}
-  for m in ts.finditer(fstring):
-    if m[1]:
-      phs = m[1].lower()
-      u['state'] = phs
-  for n in ctmd.finditer(fstring):
-    if n[1]:
-      u[n[1]] = n[2]
-    else:
-      u[n[3]] = n[4]
-  return u
-
-f = open(cs_loc, 'w')
-f.write("")
-f.close()
+    if debug:
+      print(f"Found file at {fifo_path}")
+    if os.path.isfile(fifo_path): # pipes return false for isfile
+      print(f"Error: file at {fifo_path} is not a fifo")
+      sys.exit(1)
```

Comment:
Make sure this is rebased on main and replace most instances of print with the appropriate [logging level](https://docs.python.org/3/library/logging.html#logging-levels)

Import or create a logger for each file like [this](https://github.com/micro-nova/AmpliPi/pull/661/files#diff-6c7221b76e99dbda44da1b28ea9905978bc07da6bead60b39df334570459e086R84-R85), using `getLogger()` and supplying a name to the logger. If you provide the same name as elsewhere, it will be the same logger and setting the logging level will be unnecessary.